src/WebJobs.Script.WebHost/TokenExpiration/TokenExpirationService.cs

```diff
@@ -0,0 +1,178 @@
+        private readonly WebJobsScriptHostService _scriptHost;
+        private CancellationTokenSource _cancellationTokenSource;
+        private Task _analysisTask;
+        private bool _disposed;
+        private bool _analysisScheduled;
+        private ILogger _logger;
+
+        public TokenExpirationService(IEnvironment environment, WebJobsScriptHostService scriptHost, ILoggerFactory loggerFactory, IOptionsMonitor<StandbyOptions> standbyOptionsMonitor)
+        {
+            _environment = environment;
+            _scriptHost = scriptHost;
+            _loggerFactory = loggerFactory;
+            _standbyOptionsMonitor = standbyOptionsMonitor;
+            _logger = _loggerFactory.CreateLogger<TokenExpirationService>();
+        }
+
+        public Task StartAsync(CancellationToken cancellationToken)
+        {
+            try
+            {
+                if (_environment.IsCoreTools())
+                {
+                    return Task.CompletedTask;
+                }
+                if (_standbyOptionsMonitor.CurrentValue.InStandbyMode)
+                {
+                    _standbyOptionsMonitor.OnChange(standbyOptions =>
+                    {
+                        if (!standbyOptions.InStandbyMode && !_analysisScheduled)
+                        {
+                            ScheduleTokenExpirationCheck();
+                        }
+                    });
+                }
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError(ex, "Error starting token expiration service. Handling error and continuing.");
+            }
+
+            return Task.CompletedTask;
+        }
+
+        public Task StopAsync(CancellationToken cancellationToken)
+        {
+            try
+            {
+                _cancellationTokenSource?.Cancel();
+
+                if (_analysisTask != null && !_analysisTask.IsCompleted)
+                {
+                    _logger.LogDebug("Token expiration service stopped before analysis completion. Waiting for cancellation.");
+
+                    return _analysisTask;
+                }
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError(ex, "Error stopping Token expiration service. Handling error and continuing.");
+            }
+            return Task.CompletedTask;
+        }
+
+        private void ScheduleTokenExpirationCheck()
+        {
+            _analysisScheduled = true;
+            _cancellationTokenSource = new CancellationTokenSource();
+
+            _analysisTask = Task.Delay(TimeSpan.FromSeconds(5), _cancellationTokenSource.Token)
+               .ContinueWith(t => AnalyzeSasTokenInUri());
+        }
+
+        private void AnalyzeSasTokenInUri()
+        {
+            var jobHost = _scriptHost.GetService<IScriptJobHost>();
+
+            if (jobHost == null
+                || _cancellationTokenSource.IsCancellationRequested)
```

Comment:
nit: this is short enough that is can on on the same line

```suggestion
            if (jobHost == null || _cancellationTokenSource.IsCancellationRequested)
```

And if `is` is available `jobHost is null` is cleaner